workflow/template_validator_test.go

```diff
@@ -233,17 +233,37 @@ func TestValidateTemplate(t *testing.T) {
 
 func TestRenderTemplate(t *testing.T) {
 	tests := []struct {
-		name          string
-		hwAddress     []byte
-		expectedError func(t *testing.T, err error)
+		name             string
+		hwAddress        []byte
+		templateID       string
+		templateData     string
+		skip             string
+		expectedError    func(t *testing.T, err error)
+		expectedTemplate string
 	}{
 		{
-			name:      "valid-hardware-address",
-			hwAddress: []byte("{\"device_1\":\"08:00:27:00:00:01\"}"),
+			name:         "valid-hardware-address",
+			hwAddress:    []byte("{\"device_1\":\"08:00:27:00:00:01\"}"),
+			templateID:   "49748301-d0d9-4ee9-84df-b64e6e1ef3dd",
+			templateData: validTemplate,
+			expectedTemplate: `
+version: "0.1"
+name: hello_world_workflow
+global_timeout: 600
+tasks:
+  - name: "hello world"
+    worker: "08:00:27:00:00:01"
+    actions:
+    - name: "hello_world"
+      image: hello-world
+      timeout: 60
+`,
 		},
 		{
-			name:      "invalid-hardware-address",
-			hwAddress: []byte("{\"invalid_device\":\"08:00:27:00:00:01\"}"),
+			name:         "invalid-hardware-address",
+			templateData: validTemplate,
+			skip:         "Not sure if this is an expected error or not but this test is not well written",
```

Comment:
Why do we want to skip this test?

Reply:
the error does not happen with the current code

Reply:
That's odd because I do get an error:

```
/ # cat tmp/hw.json | tink hardware push
2021/01/13 08:16:06 Hardware data pushed successfully

/ # cat tmp/hello.tmpl | tink template create -n hello
Created Template:  a6aff6d5-5577-11eb-aef3-0242ac120004
 
/ # tink workflow create -t a6aff6d5-5577-11eb-aef3-0242ac120004 -r '{"invalid_device": "08:00:27:00:00:01"}'
2021/01/13 08:24:04 rpc error: code = Unknown desc = failed to render template, invalid hardware address: {"invalid_device": "08:00:27:00:00:01"}

/ # cat tmp/hello.tmpl 
version: "0.1"
name: hello_world_template
global_timeout: 6
tasks:
  - name: "hello world"
    worker: "{{.device_1}}"
    actions:
      - name: "hello_world"
        image: hello-world
        timeout: 60
/ # 

```

Reply:
The error is not caused by a particular validation on the field, but it is caused by the `html/template` package. That is wrong

Reply:
Fixed! Thanks